lib/protobuf/optionable.rb

```diff
@@ -1,23 +1,59 @@
-require 'active_support/concern'
-
 module Protobuf
   module Optionable
-    extend ::ActiveSupport::Concern
-
     module ClassMethods
       def get_option(name)
-        @_optionable_options.try(:[], name)
+        name = name.to_s
+        option = optionable_descriptor_class.get_field(name, true)
+        fail ArgumentError, "invalid option=#{name}" unless option
+        unless option.fully_qualified_name.to_s == name
+          # Eventually we'll deprecate the use of simple names of fields completely, but for now make sure people
+          # are accessing options correctly. We allow simple names in other places for backwards compatibility.
+          fail ArgumentError, "must access option using its fully qualified name: #{option.fully_qualified_name.inspect}"
+        end
+        if @_optionable_options.try(:key?, name)
+          value = @_optionable_options[name]
+        else
+          value = option.default_value
+        end
+        if option.type_class < ::Protobuf::Message
```

Comment:
Yeah that's because B.class == Class, @embark is right here

> On Apr 12, 2016, at 4:31 PM, embark <notifications@github.com> wrote:
> 
> In lib/protobuf/optionable.rb:
> 
> >      module ClassMethods
> >        def get_option(name)
> > -        @_optionable_options.try(:[], name)
> > +        name = name.to_s
> > +        option = optionable_descriptor_class.get_field(name, true)
> > +        fail ArgumentError, "invalid option=#{name}" unless option
> > +        unless option.fully_qualified_name.to_s == name
> > +          # Eventually we'll deprecate the use of simple names of fields completely, but for now make sure people
> > +          # are accessing options correctly. We allow simple names in other places for backwards compatibility.
> > +          fail ArgumentError, "must access option using its fully qualified name: #{option.fully_qualified_name.inspect}"
> > +        end
> > +        if @_optionable_options.try(:key?, name)
> > +          value = @_optionable_options[name]
> > +        else
> > +          value = option.default_value
> > +        end
> > +        if option.type_class < ::Protobuf::Message
> I can switch to a ternary operator, but the is_a? behavior won't work on a subclass of ::Protobuf::Message:
> 
> class A; end
> class B < A;end
> B.is_a? A 
> => false
> I think the < operator is the correct method to use here
> 
> ―
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
>